### ISSUE TITLE
One letter search doesn't work

### ISSUE THREAD

**Issue by jgbneatdesign** [teamtnt/tntsearch#22]
Hello! Thanks for your package. It's really great.

I have it setup in a Laravel 5.2 application. The main issue I have is with the Ajax search. Doing search with one letter is necessary for my app, just like Google. After setting the asYouType property, which by the way needs the initialization of the class otherwise it won't accept it, it returns results for one letter but the wrong way. 

Say I have many posts that have the word yeah in their title, when I type 'y' it will just show on post. But 'yeah' returns all the posts. That's a big issue for me. 

Please see what you can do for that. And again good job! 

- Regards!

**Comment by nticaric** [comment]
Thank you!

I'm not sure I understand the problem you have? So the search returns only 1 result, but it should return multiple?

**Comment by jgbneatdesign** [comment]
Yes exactly. I have many posts that have that one letter in their title but
I got one post returned in the result while it should give me a collection.

On Jul 3, 2016 4:14 AM, "Nenad Ticaric" <notifications@github.com> wrote:

Thank you!

I'm not sure I understand the problem you have? So the search returns only
1 result, but it should return multiple?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/teamtnt/tntsearch/issues/22#issuecomment-230143610>,
or mute
the thread
<https://github.com/notifications/unsubscribe/AHrcKPpDvDjhyJ-NwgJgKjopvWpcz4U-ks5qR32JgaJpZM4JDx_Q>
.

**Comment by nticaric** [comment]
The `asYouType` functionality will only include words that start with a particular letter, so for example if you type the letter `y` it will search the index for the most common word which starts with `y` and return that word. Let's say that the most common word is `yes`. After that it will return all documents that contain the word `yes`. So if the result set returns only one result, this means that the word `yes` was found only in one document

**Comment by jgbneatdesign** [comment]
This would make sense only if there was only one document with the proposed
title yes. But I have more than one documents with yes (as proposed) within
their title. Shouldn't the asYouType property return a collection of
documents instead of one?

**Comment by nticaric** [comment]
Yes, it should return a collection if more than one occurrences are found. Have you made sure that you have the latest version of TNTSearch?

**Comment by jgbneatdesign** [comment]
Yes it should but it doesn't. I have the latest version of the package.
0.6. And apart from this issue, as suggested by another user you should
implement fuzzy searching as soon as possible.

**Comment by nticaric** [comment]
Can you write a failing unit test for this so I can try to fix it?

**Comment by jgbneatdesign** [comment]
No I can't.  Because I don't know how to test my code yet. But I 'll learn
that. And for now what I can do is send you some screenshots or gists of my
code and the results I'm getting. If it's good for you let me know.

**Comment by nticaric** [comment]
Ok, lets try that. Is it possible to send me the document collection also, so I can recreate it. And also which particular letter/word you're trying

**Comment by jgbneatdesign** [comment]
Yes I will do my best for that. Thanks!

**Comment by jgbneatdesign** [comment]
Here are my search controller, 2 models, 2 db files and 1 command to index
them. I'm sorry for any inconvenience those might be. Regards.

**Comment by nticaric** [comment]
It seems like you forgot to attach the files

**Comment by jgbneatdesign** [comment]
I'm sorry. Please see here:
https://drive.google.com/folderview?id=0B4Hp3qR50YdmcFZmblB6U2ZfZDA

**Comment by nticaric** [comment]
Ok, after looking at your index here's the issue for the word `yeah`. In your mp3 collection you have also a word `yes` which occurs only once in all of the documents and when you type only `y` it will match the word `yes` since the levenshtein distance for `y` is closer to the word `yes` than to `yeah`

**Comment by jgbneatdesign** [comment]
Thanks for pointing this out! I will try with other keywords. By the way
I'm having some issues to get the package to work on a pc with Ubuntu on it
now. It returns empty results. I'm not sure if this has to do with folder
permission or something. I was on Mac before that. But I guess I must open
a new issue on Github.
